src/scripts/antithesis.zig

```diff
@@ -0,0 +1,236 @@
+//! Deployment script for our systest (src/testing/systest).
+//!
+//! * Builds the Java client and the associated workload using Maven
+//! * Builds Docker images for the workload, replicas, and config
+//! * Optionally pushes the images to the Antithesis registry
+//!
+//! Currently there's no support for triggering tests with this script.
+
+const std = @import("std");
+const builtin = @import("builtin");
+const log = std.log;
+const assert = std.debug.assert;
+
+const stdx = @import("../stdx.zig");
+const flags = @import("../flags.zig");
+const fatal = flags.fatal;
+const Shell = @import("../shell.zig");
+
+pub const CLIArgs = struct {
+    // Docker tag to build (and possibly push)
+    tag: []const u8,
+    // Whether to push the built tag to the Antithesis registry
+    push: bool = false,
+};
+
+const Image = enum { config, workload, replica };
+
+pub fn main(shell: *Shell, _: std.mem.Allocator, cli_args: CLIArgs) !void {
+    assert(try shell.exec_status_ok("docker --version", .{}));
+
+    assert(cli_args.tag.len > 0);
+    assert(std.mem.indexOfAny(u8, cli_args.tag, &std.ascii.whitespace) == null);
+
+    try shell.zig("build -Drelease", .{});
+
+    // Build Java client library
+    {
+        try shell.pushd("./src/clients/java");
+        defer shell.popd();
+
+        try shell.exec("mvn clean install --batch-mode --quiet -Dmaven.test.skip", .{});
+    }
+
+    // Build workload
+    {
+        try shell.pushd("./src/testing/systest/workload");
+        defer shell.popd();
+
+        try shell.exec("mvn clean package --batch-mode --quiet", .{});
+    }
+
+    const images = comptime std.enums.values(Image);
+    inline for (images) |image| {
+        try build_image(shell, image, cli_args.tag);
+        if (cli_args.push) {
+            try push_image(shell, image, cli_args.tag);
+        }
+    }
+}
+
+fn build_image(
+    shell: *Shell,
+    comptime image: Image,
+    tag: []const u8,
+) !void {
+    switch (image) {
+        .config => {
+            const image_dir = try shell.exec_stdout("mktemp -d", .{});
+
+            try shell.pushd(image_dir);
+            defer shell.popd();
+
+            // TODO(owickstrom): remove the need for .env file by rendering docker-compose.yaml
+            // with the correct tag?
+            const env_file = try shell.fmt("{s}/.env", .{image_dir});
+
+            const env_file_contents = try shell.fmt("TAG={s}", .{tag});
+            _ = try shell.file_ensure_content(env_file, env_file_contents);
+
+            const docker_compose_file = try shell.fmt("{s}/docker-compose.yaml", .{
+                image_dir,
+            });
+            _ = try shell.file_ensure_content(docker_compose_file, docker_compose_contents);
+
+            try shell.exec("mkdir -p volumes/database", .{});
```

Comment:
there's `Dir.makePath`

In general, don't gratitiously shell out to posix utilities, minimize OS dependencies. Zig stdlib is our interface to OS facilities.